execution-service/src/main/java/org/ihtsdo/rvf/execution/service/whitelist/AcceptanceGatewayClient.java

```diff
@@ -0,0 +1,86 @@
+    private String acceptanceGatewayServiceUrl;
+    private RestTemplate restTemplate;
+    private HttpHeaders headers;
+    private Resty resty;
+    private static final String ALL_CONTENT_TYPE = "*/*";
+    private static final ParameterizedTypeReference<List<WhitelistItem>> WHITELIST_ITEM_LIST_TYPE_REFERENCE = new ParameterizedTypeReference<List<WhitelistItem>>() {};
+
+    protected static Gson gson;
+    static {
+        GsonBuilder gsonBuilder = new GsonBuilder();
+        gsonBuilder.setPrettyPrinting();
+        gsonBuilder.disableHtmlEscaping();
+        gsonBuilder.excludeFieldsWithoutExposeAnnotation();
+        gson = gsonBuilder.create();
+    }
+
+    public AcceptanceGatewayClient(String acceptanceGatewayServiceUrl, String authToken) throws URISyntaxException, IOException {
+        this.acceptanceGatewayServiceUrl = acceptanceGatewayServiceUrl;
+
+        resty = new Resty(new RestyOverrideAccept(ALL_CONTENT_TYPE));
+        resty.withHeader("Cookie", authToken);
+        resty.withHeader("Connection", "close");
+        resty.authenticate(this.acceptanceGatewayServiceUrl, null,null);
+
+        headers = new HttpHeaders();
+        headers.add("Cookie", authToken);
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        restTemplate = new RestTemplateBuilder()
+                .rootUri(this.acceptanceGatewayServiceUrl)
+                .additionalMessageConverters(new GsonHttpMessageConverter())
+                .errorHandler(new ExpressiveErrorHandler())
+                .build();
+
+        //Add a ClientHttpRequestInterceptor to the RestTemplate to add cookies as required
+        restTemplate.getInterceptors().add(new ClientHttpRequestInterceptor(){
+            @Override
+            public ClientHttpResponse intercept(HttpRequest request, byte[] body, ClientHttpRequestExecution execution) throws IOException {
+                request.getHeaders().addAll(headers);
+                return execution.execute(request, body);
+            }
+        });
+    }
+
+    public static AcceptanceGatewayClient createClient(String acceptanceGatewayServiceUrl, String imsUrl, String username, String password) {
+        IMSRestClient imsClient = new IMSRestClient(imsUrl);
+        String token = null;
+        try {
+            token = imsClient.loginForceNewSession(username, password);
+            return new AcceptanceGatewayClient(acceptanceGatewayServiceUrl, token);
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (URISyntaxException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    public List<WhitelistItem> getWhitelistItemsByAssertionIds(Set<String> assertionIds) {
+        ResponseEntity<List<WhitelistItem>> responseEntity = restTemplate.exchange(this.acceptanceGatewayServiceUrl + "/whitelist-items/validation-rules", HttpMethod.POST, new org.springframework.http.HttpEntity<>(assertionIds), WHITELIST_ITEM_LIST_TYPE_REFERENCE);
```

Comment:
I thought you would be using the `/bulk-validate` endpoint here to check if the failures from the current RVF report are whitelisted or not?
Loading all whitelisted items from AAG into the RVF and checking this side seems unnecessarily heavy and memory intensive?